### DISCUSSION TITLE
Best way to nest resources

### DISCUSSION THREAD

**Discussion by leo-unglaub** [filamentphp/filament#473] [General]
Hey,
i am trying to do the following. I have the Resource `Foobar` and under there i have `Foo` and `Bar` as relationships. This works great, but now i have the problem that `Bar` itself has a relationship to `Yeah`. So basically a third level. Is there a way of doing that in filament? I looked thru the docs, but i did not find a way of doing that.

Do you have any pointers into the right direction for me?
Thanks so much and thanks for the awesome work on filament!
Greetings
Leo

**Comment by martinmildner** [comment]
I just got into the same problem with multiple nested relationships. Not sure about the best way to solve it.
For now I am overwriting the Edit-RecordAction in the ```BarRelationManager``` file so that it opens the edit page of ```BarResource```:

```
->recordActions([
  Link::make('')
    ->label('Edit')
    ->url(fn ($record) => BarResource::generateUrl('edit', $record))
])
```

The disadvantage is that the ```BarResource``` "Cancel" button of course still leads back to the ```BarResource``` index page instead of the ```Foobar``` index page which can be very confusing for users.

Has anyone found another solution?

**Reply by hrvoj3e** [reply]
Is `Link::make()` a custom component you made or ?
